libs/game-state/src/lib/related-cards/dynamic-pools.ts

```diff
@@ -455,16 +931,21 @@ const canBeDiscoveredByClass = (card: ReferenceCard, currentClass: string): bool
 	return card.classes.includes(currentClass.toUpperCase()) || card.classes.includes(CardClass[CardClass.NEUTRAL]);
 };
 
-const fromClass = (card: ReferenceCard, playerClass: CardClass): boolean => {
-	return card?.classes?.includes(CardClass[playerClass]) ?? false;
-};
-
 const fromAnotherClass = (card: ReferenceCard, currentClass: string): boolean => {
 	return (
 		!card?.classes?.includes(CardClass[CardClass.NEUTRAL]) && !card?.classes?.includes(currentClass?.toUpperCase())
 	);
 };
 
+
+
+const getPlayerOrOpponent = (deckState: DeckState, gameState: GameState): PlayerGameState | undefined => {
```

Comment:
Small personal preference: the "fullGameState" today is a crutch, caused by the way the project is architected (and the fact that there is no proper multi-threading in the browser), so I would just make it obvious that this method reads from that game state, like `getPlayerOrOpponentFromFullGameState`